wsmaster/codenvy-hosted-platform-api-impl/src/test/java/com/codenvy/api/workspace/LimitsCheckingWorkspaceManagerTest.java

```diff
@@ -365,9 +424,12 @@ public void shouldBeAbleToCreateWorkspaceWithMultipleMachinesIncludingMachineWit
 
     @Test
     public void shouldBeAbleToStartWorkspaceWithMultipleMachinesIncludingMachineWithoutLimitsWhichDoesNotExceedRamLimit() throws Exception {
+        when(systemRamInfoProvider.getSystemRamInfo()).thenReturn(new SystemRamInfo(0, parseSize("3 GiB")));
         LimitsCheckingWorkspaceManager manager = spy(new LimitsCheckingWorkspaceManager(2,
                                                                                         "3gb",
                                                                                         "1gb", // <- workspaces ram limit
+                                                                                        0,
+                                                                                        systemRamInfoProvider,
                                                                                         null,
```

Comment:
I don't see any test that check logic around semaphore usage. @vinokurig  Can you point out where these tests are?